client/src/components/profile/profile.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import './profile.css';
+import Spinner from "react-spinner-material";
+import cookie from 'react-cookie';
+
+
+export class Profile extends React.Component {
+state = {
+  mode: 'view',
+  userProfile: null,
+  searchStatus: 'idle'
+}
+
+componentDidMount(){
+  let udetails = cookie.load('udetails');
+  if(udetails){
+    udetails = JSON.parse(window.atob(udetails.split('.')[1])).u$u;
```

Comment:
What does atob mean?

Reply:
I use it to avoid making request and wait for response. It is used for decoding base64.
So instead of making request and reading cookie from the request then decode it using jwt module and send it back in response as plain text, I just decode it in front-end. Aw yeah I'm verifying the cookie using the /checkAuth endpoint.